js/components/screenGroupDetails.js

```diff
@@ -17,20 +40,33 @@ window.ScreenGroupDetails = React.createClass({
     }
     return name;
   },
+  renderFinishButton: function () {
+    return (
+      <a
+        className="btn btn-positive btn-block"
+        onClick={this.handleSubmitGroup}
+      >Finish</a>
+    );
+  },
   renderGroupRules: function () {
     var groupId = this.props.groupId;
-    var rules;
     var i;
     var groupJson = this.props.getGroups();
     for (i = 0; i < groupJson.length; i++) {
       if (groupJson[i].id === groupId) {
-        rules = groupJson[i].groupRules;
+        this.setState({
+          rules: groupJson[i].groupRules
```

Comment:
I read the link you provied  and [https://viveksrao.com/2016/01/07/react-life-cycle/](url).
As suggested by you 

> Either change the state before the render or after the render. But not during the render.

I think I should use `componentDidMount`

>`componentDidMount` 
When – Runs immediately after render.
Why – By the time this function is called, the component’s DOM exists. So this is a handy spot for integrating with other frameworks such as third party component libraries. This is also a good spot to set timers and make AJAX requests since you now know the component is rendered in the DOM.

Should I?